### PR TITLE
cache fix for MongoDb

### PR DESCRIPTION
**Pull request by susmithavani30**

### CODE REVIEW

File: copper-server-impl/src/main/java/com/pogeyan/cmis/impl/factory/DatabaseServiceFactory.java [hunk 1]
```diff
@@ -49,4 +46,16 @@ public static void add(IDBClientFactory dbFactoryInstance) {
...
+
+	public static void close(String repoId) {
+		if (repoId != null) {
+			getInstance(repoId).close(repoId);
+		}
+	}
+
+	public static void closeAll() {
+		dbFactory.values().forEach(dbInstance -> {
+			dbInstance.close(null);
```
Comment (fahadsuhaib) [inline copper-server-impl/src/main/java/com/pogeyan/cmis/impl/factory/DatabaseServiceFactory.java:58]: We should use the closeAll API here?

File: copper-server-mongo/src/main/java/com/pogeyan/cmis/data/mongo/services/MongoClientFactory.java [hunk 2]
```diff
@@ -84,8 +89,22 @@
 	private static String MRELATIONOBJECCTDAOIMPL = "MRelationObjectDAOImpl";
 	private Map<Class<?>, String> objectServiceClass = new HashMap<>();
 	private final Map<String, Datastore> clientDatastores = new HashMap<String, Datastore>();
-	private final Map<String, MongoClient> mongoClient = new HashMap<String, MongoClient>();
 	private Morphia morphia = new Morphia();
+	private final static Cache<String, MongoClient> mongoClient;
+	private static RemovalListener<String, MongoClient> removalListener = new RemovalListener<String, MongoClient>() {
+		public void onRemoval(RemovalNotification<String, MongoClient> removal) {
+			MongoClient conn = removal.getValue();
+			LOG.info("Closing MongoClient connection in RemovalListener");
```
Comment (fahadsuhaib) [inline copper-server-mongo/src/main/java/com/pogeyan/cmis/data/mongo/services/MongoClientFactory.java:97]: Add repoId in the log

File: copper-server-mongo/src/main/java/com/pogeyan/cmis/data/mongo/services/MongoClientFactory.java [hunk 3]
```diff
@@ -84,8 +89,22 @@
...
 	private Map<Class<?>, String> objectServiceClass = new HashMap<>();
 	private final Map<String, Datastore> clientDatastores = new HashMap<String, Datastore>();
-	private final Map<String, MongoClient> mongoClient = new HashMap<String, MongoClient>();
 	private Morphia morphia = new Morphia();
+	private final static Cache<String, MongoClient> mongoClient;
+	private static RemovalListener<String, MongoClient> removalListener = new RemovalListener<String, MongoClient>() {
+		public void onRemoval(RemovalNotification<String, MongoClient> removal) {
+			MongoClient conn = removal.getValue();
+			LOG.info("Closing MongoClient connection in RemovalListener");
+			conn.close();
```
Comment (fahadsuhaib) [inline copper-server-mongo/src/main/java/com/pogeyan/cmis/data/mongo/services/MongoClientFactory.java:98]: Why are we not removing the clientDatastore from this?

Comment (susmithavani30) [reply copper-server-mongo/src/main/java/com/pogeyan/cmis/data/mongo/services/MongoClientFactory.java:98]: we are removing in close function

**Review by fahadsuhaib — APPROVED** [review]

**Pull request merged**